olive/evaluator/metric_config.py

```diff
@@ -2,22 +2,22 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 # --------------------------------------------------------------------------
-from pathlib import Path
 from typing import Callable, List, Union
 
 from pydantic import validator
 
 from olive.common.config_utils import ConfigBase, ConfigParam, create_config_class
+from olive.resource_path import OLIVE_RESOURCE_ANNOTATIONS
 
 WARMUP_NUM = 10
 REPEAT_TEST_NUM = 20
 SLEEP_NUM = 0
 
-
+user_path_config = ["script_dir", "data_dir", "user_script"]
 _common_user_config = {
-    "script_dir": ConfigParam(type_=Union[Path, str]),
-    "user_script": ConfigParam(type_=Union[Path, str]),
-    "data_dir": ConfigParam(type_=Union[Path, str]),
+    "script_dir": ConfigParam(type_=OLIVE_RESOURCE_ANNOTATIONS, is_path=True),
+    "user_script": ConfigParam(type_=OLIVE_RESOURCE_ANNOTATIONS, is_path=True),
```

Comment:
>user_script [](http://example.com/codeflow?start=5&length=11)

do we need to make the user_script and script_dir consistent with olive pass one?

Reply:
Updated.